### PR TITLE
Allow configuring the trace source logger without a TraceListener ins…

### PR DESCRIPTION
**Pull request by moozzyk**
…tance

Addresses parking lot/I-#405

### CODE REVIEW

File: src/Microsoft.Extensions.Logging.TraceSource/TraceSourceLoggerProvider.cs [hunk 1]
```diff
@@ -115,7 +119,10 @@ public void Dispose()
             {
                 _disposed = true;
                 _rootTraceListener.Flush();
-                _rootTraceListener.Dispose();
+                if (_rootTraceListener != null)
```
Comment (davidfowl) [inline src/Microsoft.Extensions.Logging.TraceSource/TraceSourceLoggerProvider.cs]: This is so old school

```C#
_rootTraceListener?.Dispose();
```

💃

**Comment by davidfowl** [comment]
:shipit:

**Comment by cesarbs** [comment]
:+1:

File: src/Microsoft.Extensions.Logging.TraceSource/TraceSourceLoggerProvider.cs (hunk 1 above)
Comment (moozzyk) [inline src/Microsoft.Extensions.Logging.TraceSource/TraceSourceLoggerProvider.cs]:
```
10 PRINT "I AM NOT OLDSCHOOL"
20 GOTO 10
```

**Pull request closed without merging**